packages/core/logger/src/index.js

```diff
@@ -33,137 +32,106 @@ export class Logger {
       }));
     };
     this.logFile = options.logFile;
-    this.context = options.context;
-  }
-
-  /**
-   * Parses the logFile, returning an array of JSON objects containing the
-   * log messages.
-   * @param {Number} limit specifies how many log messages to return from the
-   *  end of the log file
-   * @returns {Array} array containing
-   *  - msg: the log message
-   *  - logLevel: log level (ie 'info', 'debug')
-   *  - name: process name (always "embark")
-   *  - timestamp: timestamp of log message (milliseconds since 1/1/1970)
-   */
-  parseLogFile(limit) {
-    let matches;
-    let logs = [];
-    const logFile = fs.readFileSync(this.logFile, 'utf8');
-    while ((matches = LOG_REGEX.exec(logFile)) !== null) {
-     // This is necessary to avoid infinite loops with zero-width matches
-     if (matches.index === LOG_REGEX.lastIndex) {
-       LOG_REGEX.lastIndex++;
-     }
-
-     if (matches && matches.length) {
-       logs.push({
-         msg: [matches[3]],
-         logLevel: matches[2],
-         name: 'embark',
-         timestamp: date.parse(matches[1], DATE_FORMAT).getTime()
-       });
-     }
-    }
 
-    // if 'limit' is specified, get log lines from the end of the log file
-    if(limit && limit > 0 && logs.length > limit){
-      logs.slice(limit * -1);
-    }
-    return logs;
+    this.writeToFile = async.cargo((tasks, callback) => {
+      if (!this.logFile) {
+        return callback();
+      }
+      let logs = '';
+      let origin = "[" + ((new Error().stack).split("at ")[3]).trim() + "]";
+      tasks.forEach(task => {
+        logs += `[${date.format(new Date(), DATE_FORMAT)}] ${task.prefix} ${task.args}\n`;
+      });
+      const formattedDate = [`[${date.format(new Date(), DATE_FORMAT)}]`]; // adds a timestamp to the logs in the logFile
+      this.fs.appendFile(this.logFile, "\n" + formattedDate.concat(origin, logs).join(' '), err => {
+        if (err) {
+          this.logFunction(`There was an error writing to the log file: ${err}`, 'red');
```

Comment:
Was considering that too but then I figured that it might cause Embark to stop (needs to be verified though) which might not be what we want. Thought it would be alright to log the error but not actually emitting it to ensure embark keeps running.

Happy to change this though.